themes/default/content/learn/abstraction-encapsulation/encapsulation/index.md

```diff
@@ -24,6 +24,23 @@ Breaking apart Pulumi code is essential to keeping a clean, well-organized, and
 
 Let's take a fairly uncomplicated piece of Pulumi code: the definition of an s3 bucket:
 
+{{< chooser language "typescript,python" />}}
+
+{{% choosable language typescript %}}
+
+```typescript
+import * as pulumi from "@pulumi/pulumi";
+import * as aws from "@pulumi/aws";
+
+const bucket = new aws.s3.Bucket("my-bucket");
+
+export const bucketName = bucket.id;
+```
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
 ```python
 import pulumi
 import pulumi_aws_native as aws_native
```

Comment:
Out of curiosity, why did we choose AWS Native here, and not just AWS Classic? I went with just-Classic for the TypeScript example, and only realized later that the Python example used both. Intentional? If so, why?

Reply:
Trying to future-proof a bit and demonstrate that you can use them side-by-side, mainly. It's a common question about blending providers, and more examples that you can are good to have.